### ISSUE TITLE
extractTranscriptSeqs, needs extra check

### ISSUE THREAD

**Issue by Roleren** [Bioconductor/GenomicFeatures#2]
I spent 5 hours debugging extractTranscriptSeqs for faFile, GRangesList

When the GTF file and the fasta file have different naming on chromosomes, like chr1 in fasta vs 1 in gtf(which is the case for zebrafish genome danRer10), this error occurs:

```
extractTranscriptSeqs(faFile("./danRer10.fa"),grl)

Error in (function (classes, fdef, mtable)  : 
  unable to find an inherited method for function ‘scanFa’ for signature ‘"FaFile", "character"’
```

This error told me very little, so I went into source code for some hours and the suddenly realized the chromsome naming differed.

Is it possible to make a test on this, that if no seqnames match, it should stop, and give propper error, if it exists it did not catch this example at least.

**Comment by hpages** [comment]
Thanks for the report. I'll look into this.

**Comment by Roleren** [comment]
Still would love this feature, I was working on some new datasets today, and they had no seqlevels in common, would be nice to include a simple check.

if(!any(seqlevels(fastafile) %in% seqlevels(grl))) stop("no seqlevels in common")

**Comment by Roleren** [comment]
Still error on this:
extractTranscriptSeqs(ORFs, faFile)
Error in extractTranscriptSeqs(ORFs, faFile) : 
  failed to extract the exon ranges from 'transcripts' with exonsBy(transcripts,
  by="tx", ...)

Would be nice if you added something like this line: 
```
if(!any(seqlevels(transcripts) %in% seqlevels(x)))
    stop("FaFile had no matching seqlevels to ranges object")
```

**Comment by hpages** [comment]
Addressed in GenomicFetures 1.37.5 (commit bcc4a6ae55c02ee05637c0c9e81ca0c68666ff5e).

Sorry for letting this slip off the radar for so long.

H.

**Comment by Roleren** [comment]
Thank you, Hervé :)
